src/views/orb-view.ts

```diff
@@ -569,6 +573,12 @@ export class OrbView<N extends INodeBase, E extends IEdgeBase> implements IOrbVi
     this._simulator.simulate();
   }
 
+
+private _createSimulator(simulator: ISimulator | (() => ISimulator) | undefined): ISimulator {
+  if(typeof simulator === "function")  return simulator();
+  return simulator || SimulatorFactory.getSimulator();
+}
```

Comment:
Hmm, I think lint is not working for you because this should be indented + if should be wrapped in brackets. Can you check if husky is initialized and `npm run lint` / `npm run lint:fix` works.

Also, there is a cool Typescript trick you can use here:

```typescript
type ICallbackOrInstance<T> = T | (() => T);

export type IOrbViewSettingsInit<N extends INodeBase, E extends IEdgeBase> = Omit<
  ...
  instances?: {
    simulator?: ICallbackOrInstance<ISimulator>;
  };
}


private _createSimulator(simulator?: ICallbackOrInstance<ISimulator>): ISimulator {
   ...
}
```

Reply:
yeah that definitely looks nicer. is there a cleaner way to do the type check instead of
```
 if(typeof simulator === "function")  return simulator();
 ```

Reply:
I think that is the most clean way to check for the function. You could have a util function somewhere. Just as I wrote this comment to give you a link to which place is the best to have a utility function in, I found a function:

https://github.com/memgraph/orb/blob/main/src/utils/type.utils.ts#L87

You already have `isFunction` type guard check so you can use it here :)